app/src/main/java/cz/martykan/forecastie/activities/MainActivity.java

```diff
@@ -461,9 +497,28 @@ private void updateTodayWeatherUI() {
         todayHumidity.setText(getString(R.string.humidity) + ": " + todayWeather.getHumidity() + " %");
         todaySunrise.setText(getString(R.string.sunrise) + ": " + timeFormat.format(todayWeather.getSunrise()));
         todaySunset.setText(getString(R.string.sunset) + ": " + timeFormat.format(todayWeather.getSunset()));
+        todayUvIndex.setText(getString(R.string.uvindex) + ": " + getString(R.string.loading_data));
         todayIcon.setText(todayWeather.getIcon());
     }
 
+    public void updateUviUi() {
```

Comment:
Couldn't this be added to the `updateTodayWeatherUI` method? So you'd just update the main UI when you retrieved the UVI results.

Reply:
I don't think so! The UVIndex uses a different async function because the data does not come from the same request as TodayWeather. The same is valid to LongTermWeatherUI()

Reply:
You're probably right. I don't know enough about this to say anything definitively.